### PR TITLE
this new commit from class-4

### PR DESCRIPTION
**Pull request by Ismail788**

### CODE REVIEW

File: server/node-postgres-mydb/db/seeds/dev/01_song.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
+const records = require(' ../../../seed.json');
+
+exports.seed = function(knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('song').del()
+    .then(function () {
+      const songsList= [];
+      records.map(record =>{
+        songsList.push({
```
Comment (zkwsk) [inline server/node-postgres-mydb/db/seeds/dev/01_song.js]: Remember the pupose of this seed file is to _seed the songs_.

Songs are stored in an array in record.tracks.all, so you need to do a map or a forEach over this array to seed all of the songs indiviually.

Remember that the song DB table contains some data that reside under "record" in the JSON, so you have to do some re-mapping for each song.

File: server/node-postgres-mydb/db/seeds/dev/03_band_has_artist.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const songs = require(' ../../../seed.json');
+
+exports.seed = function(knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('artist').del()
+    .then(function () {
+      const band_has_artistList =[];
+    band_has_artistList.map(band_has_artist=> {
```
Comment (zkwsk) [inline server/node-postgres-mydb/db/seeds/dev/03_band_has_artist.js]: You map over an empty array? The array was instantiated on the line before.

File: server/node-postgres-mydb/db/seeds/dev/03_band_has_artist.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const songs = require(' ../../../seed.json');
```
Comment (zkwsk) [inline server/node-postgres-mydb/db/seeds/dev/03_band_has_artist.js]: You don't actually use any data from the json file in this file?

File: server/node-postgres-mydb/db/seeds/dev/04_artist.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const songs = require(' ../../../seed.json');
+
+exports.seed = function(knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('artist').del()
+    .then(function () {
+      const artistsList =[];
+    artistsList.map(artist=> {
```
Comment (zkwsk) [inline server/node-postgres-mydb/db/seeds/dev/04_artist.js]: Same problem as in `03_band_has_artist.js`:

You iterate over an empty array `artistsList` and you don't actually pull any data from the json file.

File: server/node-postgres-mydb/db/seeds/dev/05_band.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+const songs = require(' ../../../seed.json');
+
+exports.seed = function(knex, Promise) {
+  // Deletes ALL existing entries
+  return knex('band').del()
+    .then(function () {
+      const bandsList =[];
+    bands.map(band=> {
```
Comment (zkwsk) [inline server/node-postgres-mydb/db/seeds/dev/05_band.js]: Same problem again. See: `03_band_has_artist.js`

File: server/node-postgres-mydb/seed.json [hunk 6]
```diff
@@ -0,0 +1,213 @@
+[{
```
Comment (zkwsk) [inline server/node-postgres-mydb/seed.json]: We tried to gitignore this one. Somehow it still ended on github :( Now you need to do a `git rm seed.json` to get it out.

File: server/node-postgres-mydb/api/songs.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const router = express.Router();
+const queries = require('../data/queries');
+
+router.get('/', (req, res) =>{
+  queries.getAll().then(song =>{
+    res.json(song);
```
Comment (zkwsk) [inline server/node-postgres-mydb/api/songs.js]: I can't work out how you get to see you entire app by going to this endpoint. It should simply be showing json output. The route must be overridden from somewhere else?

**Review by zkwsk — CHANGES_REQUESTED** [review]

File: server/node-postgres-mydb/api/songs.js (hunk 7 above)
Comment (Ismail788) [reply server/node-postgres-mydb/api/songs.js]: I used mocha shai super test to test  api song, so have method call getAll and done method and I create my own queries in data folder and requiring the knex configrations

File: server/node-postgres-mydb/db/seeds/dev/03_band_has_artist.js (hunk 3 above)
Comment (Ismail788) [reply server/node-postgres-mydb/db/seeds/dev/03_band_has_artist.js]: yes I getting empty record and I dont know why i am not accessing the data

File: server/node-postgres-mydb/db/seeds/dev/04_artist.js (hunk 4 above)
Comment (Ismail788) [reply server/node-postgres-mydb/db/seeds/dev/04_artist.js]: yes I got it my mistake in both these tables I mapping over empty record

File: server/node-postgres-mydb/db/seeds/dev/05_band.js (hunk 5 above)
Comment (Ismail788) [reply server/node-postgres-mydb/db/seeds/dev/05_band.js]: Yes I need specify all record within the tables and add the right data in table

File: server/node-postgres-mydb/db/seeds/dev/01_song.js (hunk 1 above)
Comment (Ismail788) [reply server/node-postgres-mydb/db/seeds/dev/01_song.js]: yes I got it I need to see the mapping again

**Pull request merged**